logstash-core/lib/logstash/inputs/metrics.rb

```diff
@@ -39,7 +44,10 @@ def update(time, snapshot)
       #   - We can use a synchronization mechanism between the called thread (update method)
       #   and the plugin thread (run method)
       #   - How we handle back pressure here?
-      @queue << snapshot.to_event
+      #   - one snashot should be only one event
+      snapshot.to_events.each do |event|
```

Comment:
I wonder if makes sense to make this call somehow async? now this might mean a ton of work from the snapshot to transform to events and then iterate to inject each new event in the queue. what do you think?